.github/workflows/gas.yml

```
@@ -0,0 +1,43 @@
+name: Gas
+
+on:
+  pull_request:
+    paths:
+      - '**.sol'
+      - '**.yml'
+  push:
+    branches:
+      - main
+    paths:
+      - '**.sol'
+      - '**.yml'
```

```suggestion
    paths:
      - '**.sol'
      - '**.yml'
      - '**.toml'
      - 'lib/**'
      - '.gitmodules'
      - '.gas-snapshot'
  push:
    branches:
      - main
    paths:
      - '**.sol'
      - '**.yml'
      - '**.toml'
      - 'lib/**'
      - '.gitmodules'
      - '.gas-snapshot'
```

---

Or we can only exclude what we know is irrelevant to play it safe:

```suggestion
    paths-ignore:
      - '**.md'
      - 'broadcast/**'
      - '.vscode/**'
      - '.gitignore'
  push:
    branches:
      - main
    paths-ignore:
      - '**.md'
      - 'broadcast/**'
      - '.vscode/**'
      - '.gitignore'
```